tests/test_txt_date_picker.py

```diff
@@ -0,0 +1,244 @@
+    ('year', '2018', '2018'), ('year', '2019', '2019'), ('year', '2020', '2020'),
+    ('year', '2021', '2021'), ('year', '2022', '2022'), ('year', '2023', '2023'),
+    ('year', '2024', '2024'), ('year', '2025', '2025'), ('year', '2026', '2026'),
+    ('year', '2027', '2027'), ('year', '2028', '2028'), ('year', '2029', '2029'),
+    ('year', '2030', '2030'), ('year', '2031', '2031'), ('year', '2032', '2032'),
+    ('year', '2033', '2033'), ('year', '2034', '2034'), ('year', '2035', '2035')
+])
+def test_date_picker_selection(page, select_type, list_option_value, list_text):
+    """
+    @brief Verifies that selecting a month or a year from the date picker works correctly.
+
+    @param page The Playwright page instance.
+    @param select_type The type of selection ('month' or 'year').
+    @param list_option_value The value to be selected.
+    @param list_text The expected text representation of the selected value.
+
+    @test
+    1. Open the date picker by clicking the input field.
+    2. Determine whether to test month or year selection.
+    3. Select the desired option using the dropdown.
+    4. Verify that the selected value matches the expected value.
+    5. Verify that the displayed text matches the expected representation.
+
+    @assert The selected value should match the expected value.
+    @assert The displayed text should match the expected month or year.
+    """
+
+    txt_date = page.locator('#txtDate')
+    txt_date.click()
+
+    date_picker_selector = page.locator('.ui-datepicker-month') if select_type == 'month' else page.locator('.ui-datepicker-year')
+
+    date_picker_selector.select_option(list_option_value)
+
+    assert date_picker_selector.input_value() == list_option_value, f'Selected value is not {list_option_value}'
+
+    selected_text = date_picker_selector.locator('option:checked').text_content()
+    assert ''.join(selected_text.split()) == list_text, f'Displayed text is not {list_text}'
+
+
+@pytest.mark.smoke
+def test_full_date(page, read_file):
+    """
+    @brief Verifies that selecting a date from the date picker inputs the correct value.
+
+    @param page The Playwright page instance.
+    @param read_file The fixture that reads JSON test data files.
+
+    @test
+    1. Load test data for date selection and expected validation data.
+    2. Iterate through years, months, and days in the test data.
+    3. Open the date picker and select the appropriate year, month, and day.
+    4. Verify that the input field displays the correct date.
+    5. Validate the result against expected test validation data.
+
+    @assert The displayed date should match the expected validation date.
+    """
+
+    run_month_day_year_data = read_file('test_run_month_day_year_data.json')
+    validation_month_day_year_data = read_file('validation_month_day_year_data.json')
+
+    txt_date = page.locator('#txtDate')
+
+    current_year = datetime.now().year
```

Comment:
Extra algorithm into separate method with current_year, min_year and max_year as a parameters and use it in test